packages/block-library/src/post-comments-count/edit.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { useEntityId } from '@wordpress/core-data';
+import { useState, useEffect } from '@wordpress/element';
+import apiFetch from '@wordpress/api-fetch';
+import { addQueryArgs } from '@wordpress/url';
+
+function PostCommentsCountDisplay( { className } ) {
+	const postId = useEntityId( 'postType', 'post' );
+	const [ commentsCount, setCommentsCount ] = useState();
+	useEffect( () => {
+		const currentPostId = postId;
+		apiFetch( {
+			path: addQueryArgs( '/wp/v2/comments', {
+				post: postId,
+			} ),
+			parse: false,
+		} ).then( ( res ) => {
+			// Stale requests will have the `currentPostId` of an older closure.
```

Comment:
this seems weird

Reply:
It's a nice technique :)

Reply:
I'm having trouble understanding in which case `currentPostId` would be different than `postId` here. Can you expand a bit?

Reply:
- Request made with `postId === 1`.
- Request made with `postId === 2`.
- Request made with `postId === 1` resolves, `currentPostId === 1`, `postId === 2` so nothing happens.